addOns/client/src/main/java/org/zaproxy/addon/client/ui/PopupMenuExportClientMap.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2010 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.client.ui;
+
+import java.awt.Component;
+import java.io.File;
+import java.util.Locale;
+import javax.swing.JFileChooser;
+import javax.swing.filechooser.FileNameExtensionFilter;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.extension.ExtensionPopupMenuItem;
+import org.zaproxy.addon.client.ExtensionClientIntegration;
+import org.zaproxy.addon.commonlib.MenuWeights;
+import org.zaproxy.zap.utils.Stats;
+import org.zaproxy.zap.view.widgets.WritableFileChooser;
+
+@SuppressWarnings("serial")
+public class PopupMenuExportClientMap extends ExtensionPopupMenuItem {
+
+    private static final long serialVersionUID = 1L;
+    private static final String YAML_EXT = ".yaml";
+    public static final String STATS_EXPORT_CLIENTMAP =
+            ExtensionClientIntegration.PREFIX + ".export.clientmap";
+
+    private ExtensionClientIntegration extension;
+
+    /**
+     * Constructs a {@code PopupMenuExportClientMap} with the given label and extension.
+     *
+     * @param label the label of the menu item
+     * @param extension the extension to access the model and view, must not be {@code null}.
+     * @throws IllegalArgumentException if the given {@code extension} is {@code null}.
+     */
+    public PopupMenuExportClientMap(String label, ExtensionClientIntegration extension) {
+        super(label);
+
+        if (extension == null) {
+            throw new IllegalArgumentException("Parameter extension must not be null.");
+        }
+        this.extension = extension;
+
+        this.addActionListener(e -> performAction());
+    }
+
+    @Override
+    public boolean isEnableForComponent(Component invoker) {
+        return "treeClient".equals(invoker.getName());
+    }
+
+    private void performAction() {
+        File file = getOutputFile();
+        if (file == null) {
+            return;
+        }
+
+        writeClientMap(file);
+        Stats.incCounter(STATS_EXPORT_CLIENTMAP);
```

Comment:
This should be in the exporter, if you want to keep this one too it should be clear it's for the GUI.